<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Nanopore_16S_analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="nanopore_files/libs/clipboard/clipboard.min.js"></script>
<script src="nanopore_files/libs/quarto-html/quarto.js"></script>
<script src="nanopore_files/libs/quarto-html/popper.min.js"></script>
<script src="nanopore_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="nanopore_files/libs/quarto-html/anchor.min.js"></script>
<link href="nanopore_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="nanopore_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="nanopore_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="nanopore_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="nanopore_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Nanopore_16S_analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="analysing-nanopore-16s-rrna-sequences" class="level2">
<h2 class="anchored" data-anchor-id="analysing-nanopore-16s-rrna-sequences">Analysing nanopore 16S rRNA sequences</h2>
<p>Data source is from article “RESCUE: a validated Nanopore pipeline to classify bacteria through long-read, 16S-ITS-23S rRNA sequencing.” by Petrone et. al (2023). Sequences were downloaded from SRA and grouped into three.</p>
<p>Link to article: <code>https://doi.org/10.3389/fmicb.2023.1201064</code></p>
<p>Zymo research mock community sample and gut mock community each contain 4 replicate samples. Oral microbiome set contains 20 saliva samples.</p>
<section id="preprosessing" class="level3">
<h3 class="anchored" data-anchor-id="preprosessing">Preprosessing</h3>
<p>All reads were trimmed with cutadapt using the 1492R primer sequence</p>
<ul>
<li>cutadapt -g CGGTTACCTTGTTACGACTT –rc -o file input_file</li>
</ul>
<p><em>Note, that sequences are reverse complement and 16S rRNA sequence starts from 3’ end</em></p>
<p><em>About 85-90 % contained correct primer sequence on default error rate 0.1.</em></p>
</section>
<section id="qiime-2-vsearch-otu-clustering-workflow" class="level3">
<h3 class="anchored" data-anchor-id="qiime-2-vsearch-otu-clustering-workflow">Qiime 2 vsearch otu clustering workflow</h3>
<p>Workflow: Importing sequences, dereplication, clustering, chimera check, chimera and singleton removal and taxa classification.</p>
<p><em>Manifest file contains all sequence files with absolute path</em></p>
<ol type="1">
<li><p>qiime tools import –type ‘SampleData[SequencesWithQuality]’&nbsp; –input-path seq/manifest&nbsp; –output-path output.qza –input-format SingleEndFastqManifestPhred33V2</p></li>
<li><p>qiime vsearch dereplicate-sequences –i-sequences seqs.qza –o-dereplicated-table dereplicated_table.qza –o-dereplicated-sequences dereplicated_sequences.qza</p></li>
<li><p>qiime vsearch cluster-features-de-novo –i-sequences dereplicated_sequences.qza –i-table dereplicated_table.qza –p-perc-identity 0.97 –o-clustered-sequences clustered_otu_sequences.qza –o-clustered-table clustered_otu_table.qza</p></li>
<li><p>qiime vsearch uchime-denovo –i-sequences clustered_otu_sequences.qza –output-dir chimeras –i-table clustered_otu_table.qza</p></li>
<li><p>qiime feature-table filter-features –i-table clustered_otu_table.qza –m-metadata-file chimeras/nonchimeras.qza —p-min-frequency 2&nbsp; –o-filtered-table&nbsp; final_otu_table.qza</p></li>
<li><p>qiime feature-table filter-seqs &nbsp; –i-data clustered_otu_sequences.qza –i-table final_otu_table.qza&nbsp; –o-filtered-data representative_sequences.qza</p></li>
<li><p>qiime feature-classifier classify-sklearn –i-classifier silva-138-99-nb-classifier.qza –i-reads representative_sequences.qza –p-read-orientation reverse –o-classification taxonomy.qza</p></li>
</ol>
</section>
<section id="qiime-2-dada-denoising-workflow" class="level3">
<h3 class="anchored" data-anchor-id="qiime-2-dada-denoising-workflow">Qiime 2 dada denoising workflow</h3>
<p>Workflow: Importing sequences, denoising and feature classification.</p>
<p>Note. Using value 1 error per 100 bp sequence produces best compromise.</p>
<ol type="1">
<li><p>qiime tools import –type ‘SampleData[SequencesWithQuality]’&nbsp; –input-path seq/manifest&nbsp; –output-path output.qza –input-format SingleEndFastqManifestPhred33V2</p></li>
<li><p>qiime dada2 denoise-single –p-max-ee 14 —p-trunc-len 0 —i-demultiplexed-sequences —output-dir dada</p></li>
<li><p>qiime feature-classifier classify-sklearn —i-classifier silva-138-99-classifier.qza —i-reads dada/representative_sequences.qza —p-read-orientation reverse —o-classification dada/taxonomy.qza</p></li>
</ol>
</section>
<section id="loading-r-libraries" class="level3">
<h3 class="anchored" data-anchor-id="loading-r-libraries">Loading R libraries</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#required libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggthemes)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="zymo-research-mock-community-samples" class="level3">
<h3 class="anchored" data-anchor-id="zymo-research-mock-community-samples">Zymo Research mock community samples</h3>
<p>Importing data from qiime 2 files</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#dada2 results</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>zymo_mock_dada <span class="ot">&lt;-</span> <span class="fu">loadFromQIIME2</span>(<span class="at">featureTableFile=</span><span class="st">"Qiime2_files/mock_community/mock_dada_asv_table.qza"</span>, <span class="at">taxonomyTableFile=</span><span class="st">"Qiime2_files/mock_community/mock_dada_taxa.qza"</span>, <span class="at">sampleMetaFile=</span><span class="st">"Qiime2_files/mock_community/meta_dada.tsv"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co">#clustering_results</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>zymo_mock_vsearch <span class="ot">&lt;-</span> <span class="fu">loadFromQIIME2</span>(<span class="at">featureTableFile=</span><span class="st">"Qiime2_files/mock_community/mock_vsearch_otu_table.qza"</span>, <span class="at">taxonomyTableFile=</span><span class="st">"Qiime2_files/mock_community/mock_vsearch_taxa.qza"</span>, <span class="at">sampleMetaFile=</span><span class="st">"Qiime2_files/mock_community/meta_clust.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Creating dataframe from zymo kit data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#create vectors and alphabetically sorted dataframe containing real mock composition</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>genus <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Bacillus"</span>, <span class="st">"Enterococcus"</span>, <span class="st">"Escherichia-Shigella"</span>, <span class="st">"Lactobacillus"</span>, <span class="st">"Listeria"</span>, <span class="st">"Pseudomonas"</span>, <span class="st">"Salmonella"</span>, <span class="st">"Staphyloccus"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>abundance <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.174</span>,<span class="fl">0.10</span>,<span class="fl">0.101</span>,<span class="fl">0.184</span>,<span class="fl">0.141</span>,<span class="fl">0.042</span>,<span class="fl">0.104</span>,<span class="fl">0.155</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>comparison <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(abundance)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(comparison) <span class="ot">&lt;-</span> genus</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>comparison <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(comparison)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Creating data frame from dada2 results</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#agglomeration to genus and calculating relabundance</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>zymo_mock_dada <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(zymo_mock_dada, <span class="at">rank=</span><span class="st">"Genus"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>zymo_mock_dada <span class="ot">&lt;-</span> <span class="fu">transformAssay</span>(zymo_mock_dada, <span class="at">assay.type=</span><span class="st">"counts"</span>,</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">method=</span><span class="st">"relabundance"</span>, <span class="at">onRankOnly=</span><span class="cn">TRUE</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#create dataframe and table alphabetically sorted</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>zm_dada <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">assays</span>(zymo_mock_dada)<span class="sc">$</span>relabundance)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>zm_dada <span class="ot">&lt;-</span> zm_dada[<span class="fu">order</span>(<span class="fu">row.names</span>(zm_dada)),]</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(zm_dada) <span class="ot">&lt;-</span> genus </span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>zm_dada <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(zm_dada)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(zm_dada, <span class="at">digits=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">rowname</th>
<th style="text-align: right;">Mock01</th>
<th style="text-align: right;">Mock02</th>
<th style="text-align: right;">Mock03</th>
<th style="text-align: right;">Mock04</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Bacillus</td>
<td style="text-align: right;">0.23</td>
<td style="text-align: right;">0.23</td>
<td style="text-align: right;">0.28</td>
<td style="text-align: right;">0.27</td>
</tr>
<tr class="even">
<td style="text-align: left;">Enterococcus</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.06</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Escherichia-Shigella</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.06</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.01</td>
</tr>
<tr class="even">
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: right;">0.19</td>
<td style="text-align: right;">0.17</td>
<td style="text-align: right;">0.19</td>
<td style="text-align: right;">0.17</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Listeria</td>
<td style="text-align: right;">0.12</td>
<td style="text-align: right;">0.12</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.16</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pseudomonas</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Salmonella</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="even">
<td style="text-align: left;">Staphyloccus</td>
<td style="text-align: right;">0.43</td>
<td style="text-align: right;">0.38</td>
<td style="text-align: right;">0.38</td>
<td style="text-align: right;">0.33</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Creating data.frame from vsearch results</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#agglomeration to genus and calculating relabundance</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>zymo_mock_vsearch <span class="ot">&lt;-</span> <span class="fu">agglomerateByRank</span>(zymo_mock_vsearch, <span class="at">rank=</span><span class="st">"Genus"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>zymo_mock_vsearch <span class="ot">&lt;-</span> <span class="fu">transformAssay</span>(zymo_mock_vsearch, <span class="at">assay.type=</span><span class="st">"counts"</span>,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">method=</span><span class="st">"relabundance"</span>, <span class="at">onRankOnly=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#create dataframe and table alphabetically sorted</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>zm_vsearch <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">assays</span>(zymo_mock_vsearch)<span class="sc">$</span>relabundance)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>zm_vsearch <span class="ot">&lt;-</span> zm_vsearch[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,]</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>zm_vsearch <span class="ot">&lt;-</span> zm_vsearch[<span class="fu">order</span>(<span class="fu">row.names</span>(zm_vsearch)),]</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(zm_vsearch) <span class="ot">&lt;-</span> genus</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>zm_vsearch <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(zm_vsearch)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(zm_vsearch, <span class="at">digits=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">rowname</th>
<th style="text-align: right;">Mock01</th>
<th style="text-align: right;">Mock02</th>
<th style="text-align: right;">Mock03</th>
<th style="text-align: right;">Mock04</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Bacillus</td>
<td style="text-align: right;">0.23</td>
<td style="text-align: right;">0.24</td>
<td style="text-align: right;">0.27</td>
<td style="text-align: right;">0.28</td>
</tr>
<tr class="even">
<td style="text-align: left;">Enterococcus</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.06</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Escherichia-Shigella</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.06</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.01</td>
</tr>
<tr class="even">
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: right;">0.16</td>
<td style="text-align: right;">0.16</td>
<td style="text-align: right;">0.17</td>
<td style="text-align: right;">0.16</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Listeria</td>
<td style="text-align: right;">0.11</td>
<td style="text-align: right;">0.11</td>
<td style="text-align: right;">0.14</td>
<td style="text-align: right;">0.15</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pseudomonas</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.01</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Salmonella</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.01</td>
</tr>
<tr class="even">
<td style="text-align: left;">Staphyloccus</td>
<td style="text-align: right;">0.36</td>
<td style="text-align: right;">0.35</td>
<td style="text-align: right;">0.34</td>
<td style="text-align: right;">0.31</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Merging data frames</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#merging dataframes and renaming columns</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>summary_df <span class="ot">&lt;-</span> <span class="fu">left_join</span>(comparison, zm_dada, <span class="at">by=</span><span class="st">"rowname"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>summary_df <span class="ot">&lt;-</span> <span class="fu">left_join</span>(summary_df, zm_vsearch, <span class="at">by=</span><span class="st">"rowname"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(summary_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Genus"</span>, <span class="st">"Real"</span>, <span class="st">"Dada01"</span>, <span class="st">"Dada02"</span>, <span class="st">"Dada03"</span>, <span class="st">"Dadao4"</span>, <span class="st">"Vsearch01"</span>, <span class="st">"Vsearch02"</span>, <span class="st">"Vsearch03"</span>, <span class="st">"Vsearch04"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>molten_summary <span class="ot">&lt;-</span> <span class="fu">melt</span>(summary_df, <span class="at">id.vars =</span> <span class="dv">1</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(molten_summary, <span class="fu">aes</span>(<span class="at">x=</span>variable,<span class="at">y=</span>value, <span class="at">fill=</span>Genus)) <span class="sc">+</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_bar</span>(<span class="at">position=</span><span class="st">"stack"</span>, <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span> <span class="fu">theme_fivethirtyeight</span>(<span class="at">base_size=</span><span class="dv">9</span>) <span class="sc">+</span> <span class="fu">scale_fill_pander</span>()</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="nanopore_files/figure-html/plot-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>In this sample set, differences between dada and vsearch look minor. However, dada looks more sensitive as two out of four samples have less variants than supposed to.</p>
<p>When compared to real composition, some genus are underpresented (Salmonella, E. coli) and some overpresented (Bacillus, Streptococcus.</p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>